src/source.cpp

```diff
@@ -0,0 +1,59 @@
+#include <capo/instance.hpp>
+#include <capo/sound.hpp>
+#include <capo/source.hpp>
+#include <impl_al.hpp>
+
+namespace capo {
+Source const Source::blank;
+
+bool Source::bind(Sound const& sound) { return valid() ? m_instance->bind(sound, *this) : false; }
+bool Source::unbind() { return valid() ? m_instance->unbind(*this) : false; }
+Sound const& Source::bound() const { return valid() ? m_instance->bound(*this) : Sound::blank; }
+
+bool Source::play() {
+	if (valid()) {
+		CAPO_CHKR(alSourcePlay(m_handle));
+		return true;
+	}
+	return false;
+}
+
+bool Source::pause() {
+	if (valid()) {
+		CAPO_CHKR(alSourcePause(m_handle));
+		return true;
+	}
+	return false;
+}
+
+bool Source::stop() {
+	if (valid()) {
+		CAPO_CHKR(alSourceStop(m_handle));
+		return true;
+	}
+	return false;
+}
+
+bool Source::loop(bool loop) {
+	if (valid()) { return detail::setSourceProp(m_handle, AL_LOOPING, loop ? AL_TRUE : AL_FALSE); }
+	return false;
+}
+
+bool Source::seek(time head) {
+	if (valid()) { return detail::setSourceProp(m_handle, AL_SEC_OFFSET, static_cast<ALfloat>(head.count())); }
+	return false;
+}
+
+bool Source::gain(float value) {
+	if (valid()) {
+		assert(value >= 0.0f);
```

Comment:
Any reason why this is fatal?

Reply:
I'll add a TODO to refactor it to send an error instead.